### DISCUSSION TITLE
Let's make this an API like Chat-GPT-to-API

### DISCUSSION THREAD

**Discussion by bi1101** [acheong08/EdgeGPT#398] [Ideas]
What do you guys think about turning this into a scalable API? Is this possible?
The potential is great as Bing seems to being GPT 4 which returns much better result than Chat GPT 3.5 turbo, even with Plus

**Comment by acheong08** [comment]
Good idea. I'll might take a shot after exams are done. Will be a bit annoying due to how Bing streams

**Reply by bi1101** [reply]
https://github.com/pavel-one/EdgeGPT-Go
There's a repo with a similar idea that may help

**Reply by bi1101** [reply]
How do you plan to do this project? I plan to do this project too with this model:

Chat-GPT-to-API (when choose gpt-4 as model) <-> custom API to convert between Bing & Open AI format <-> [EdgeGPT-Go] API <-> Bing

It's modular this way so when other repo updates they won't interfere too much.

What do you think?

**Reply by acheong08** [reply]
Yup.
I'm planning on refactoring `ChatGPT-to-API` to make it modular and be able to add different APIs easily with a generic interface and a switch case to choose different APIs based on model name. Probably gonna rename the repo to better fit the new purpose.

A few APIs planned once my exams are over

EdgeGPT, ChatGPT, Bard, and a custom search (https://github.com/acheong08/SearchAPI - work in progress).

**Reply by bi1101** [reply]
I'm assuming that these are all compatible with the openai format right? Or will they be different (Bing & Bard has their own format)

**Reply by acheong08** [reply]
Yes. The goal is for them to match OpenAI's format for use with OpenAI compatible tools
